Automattic/wp-calypso
# Membership: Enable/disable auto-renew in memberships

Related to 243-gh-Automattic/gold

## Proposed Changes

* Adds Enable/Disable auto-renew button to membership subscription view
* Adds state management
* Adds API call

## Testing Instructions

* Enable store sandbox (PCYsg-IA-p2) and point subscribe.wordpress.com at your sandbox
* Apply D135787-code
* Visit [this page](https://outdoor-squid.jurassic.ninja/2024/01/18/bronze-tier/) and purchase a subscription using a [Stripe test card](https://stripe.com/docs/testing#cards)
* Alternatively, using an account linked with Stripe, create a paid newsletter plan then create a post with access by [newsletter tier](https://jetpack.com/support/newsletter/)
* Visit the site owned by the email address you used for your subscription through Calypso live
* Visit [your purchases](https://container-heuristic-pasteur.calypso.live/me/purchases) and find the renewing membership subscription
* <img width="942" alt="Screenshot 2024-01-18 at 11 40 02 AM" src="https://github.com/Automattic/wp-calypso/assets/38750/878b6814-5321-4453-af33-3bb65526a3b0">
* Press the membership subscription to view its details
* Verify the info looks similar to the following screenshot, particularly the "Renews on" area and "Disable auto-renew"
* <img width="956" alt="Screenshot 2024-01-24 at 11 00 06 AM" src="https://github.com/Automattic/wp-calypso/assets/38750/d5ed1258-1916-4b15-a8ec-9a67605cbd8d">
* Press "Disable auto-renew"
* Verify the info updates to look similar to the following screenshot, with both areas having updated
* <img width="962" alt="Screenshot 2024-01-24 at 11 00 16 AM" src="https://github.com/Automattic/wp-calypso/assets/38750/bea92599-01d7-4375-a0a4-caf130009044">
* Press the "Back" arrow in the "Subscription Details" header
* Verify the membership subscription row was updated to show it now has an expiration
* <img width="941" alt="Screenshot 2024-01-24 at 11 00 34 AM" src="https://github.com/Automattic/wp-calypso/assets/38750/c5917d1d-00e6-4eb6-b69a-c5d8fd7547ec">
* Return to the subscription details and press "Enable auto-renew"
* Verify it has returned to its original state
* Press the "Back" arrow and verify the membership subscription row is marked renewing again
* Additionally, repeat the above test but reload the page after each update and verify it works as expected

## Updated Strings

* <img width="1079" alt="Screenshot 2024-01-25 at 6 32 19 AM" src="https://github.com/Automattic/wp-calypso/assets/38750/1b1a85d9-97d4-4f1a-a724-02408128af02">
* <img width="661" alt="Screenshot 2024-01-25 at 6 32 34 AM" src="https://github.com/Automattic/wp-calypso/assets/38750/f8b0df4b-cdb1-43f6-8be1-c50aea9f9deb">

## Pre-merge Checklist
